vizro-ai/src/vizro_ai/dashboard/nodes/data_summary.py

```diff
@@ -0,0 +1,41 @@
+"""Data Summary Node."""
+
+from typing import Dict, Tuple
+
+import pandas as pd
+from langchain_core.prompts import ChatPromptTemplate
+
+try:
+    from pydantic.v1 import BaseModel, Field
+except ImportError:  # pragma: no cov
+    from pydantic import BaseModel, Field
+
+
+def _get_df_info(df: pd.DataFrame) -> Tuple[Dict[str, str], str]:
+    """Get the dataframe schema and head info as strings."""
+    formatted_pairs = {col_name: str(dtype) for col_name, dtype in df.dtypes.items()}
+    df_sample = df.sample(5).to_markdown()
+    return formatted_pairs, df_sample
+
+
+df_sum_prompt = ChatPromptTemplate.from_messages(
+    [
+        (
+            "system",
+            """You are a data assistant with expertise naming a pandas dataframe. \n
+            Inspect the provided data and give a short unique name to the dataset. \n
+            Here is the dataframe sample:  \n ------- \n  {df_sample} \n ------- \n
+            Here is the schema:  \n ------- \n  {df_schema} \n ------- \n
+            Names currently in use: \n ------- \n {current_df_names} \n ------- \n
+            \n ------- \n
+            """,
+        ),
+        ("placeholder", "{messages}"),
+    ]
+)
+
+
+class DfInfo(BaseModel):
```

Comment:
This seems like a lot of pfaff to get a name - maybe we can similify the prompt even. I think I did some regex thing for the ids